### PR TITLE
Make resources unique by type and value

### PR DESCRIPTION
**Pull request by aelred**
- Resources are now unique by type and value
- Only resources can have a value and it must be specified
- Resources cannot have an ID specified
- Value cannot be changed

### CODE REVIEW

File: mindmaps-graql/src/main/java/io/mindmaps/graql/internal/query/InsertQueryExecutor.java [hunk 1]
```diff
@@ -219,6 +217,11 @@ private VarAdmin mergeVar(VarAdmin var) {
     private Concept putConceptByType(Optional<String> id, VarAdmin var, Type type) {
         String typeId = type.getId();
 
+        if (!type.isResourceType() && !var.getValuePredicates().isEmpty()) {
+            // TODO: Deal with this properly
```
Comment (aelred) [inline mindmaps-graql/src/main/java/io/mindmaps/graql/internal/query/InsertQueryExecutor.java]: TODO: Do TODO

File: mindmaps-graql/src/test/java/io/mindmaps/graql/reasoner/graphs/AdmissionsGraph.java [hunk 2]
```diff
@@ -275,13 +274,17 @@ private static void addRules() {
...
     }
 
-    private static Instance putEntity(EntityType type, String name) {
-        return mindmaps.putEntity(name.replaceAll(" ", "-").replaceAll("\\.", ""), type).setValue(name);
+    private static <T> void putResource(Instance instance, ResourceType<T> resourceType, T resource) {
+        Resource resourceInstance = mindmaps.putResource(resource, resourceType);
+
+        mindmaps.addRelation(hasResource)
+                .putRolePlayer(hasResourceTarget, instance)
+                .putRolePlayer(hasResourceValue, resourceInstance);
```
Comment (aelred) [inline mindmaps-graql/src/test/java/io/mindmaps/graql/reasoner/graphs/AdmissionsGraph.java]: Is this method used?

File: mindmaps-graql/src/test/java/io/mindmaps/graql/reasoner/inference/AdmissionsInferenceTest.java [hunk 3]
```diff
@@ -47,49 +47,39 @@ public static void setUpClass() {
     public void testConditionalAdmission() {
         String queryString = "match $x isa applicant; $x has admissionStatus 'conditional'";
         MatchQueryDefault query = qp.parseMatchQuery(queryString).getMatchQuery();
-        String explicitQuery = "match $x isa applicant, value 'Bob'";
-        assertQueriesEqual(reasoner.expand(query), qp.parseMatchQuery(explicitQuery).getMatchQuery());
-        assertEquals(reasoner.resolve(query), Sets.newHashSet(qp.parseMatchQuery(explicitQuery).getMatchQuery()));
+        assertQueriesEqual(reasoner.expand(query), qp.parseMatchQuery("match $x isa applicant, id 'Bob'").getMatchQuery());
```
Comment (aelred) [inline mindmaps-graql/src/test/java/io/mindmaps/graql/reasoner/inference/AdmissionsInferenceTest.java]: I think I broke these

**Comment by fppt** [comment]
@aelred travis failed because of running for too long. I think this PR is still running the scaling tests. Maybe pull from master to resolve this ?

**Pull request merged**